extensions/vertx-graphql/deployment/src/main/java/io/quarkus/vertx/graphql/deployment/VertxGraphqlProcessor.java

```diff
@@ -18,4 +34,38 @@ FeatureBuildItem feature() {
     WebsocketSubProtocolsBuildItem websocketSubProtocols(CombinedIndexBuildItem combinedIndexBuildItem) {
         return new WebsocketSubProtocolsBuildItem("graphql-ws");
     }
+
+    @BuildStep
+    List<ReflectiveClassBuildItem> registerForReflection() {
+        return Arrays.asList(
+                new ReflectiveClassBuildItem(true, true,
+                        io.vertx.ext.web.handler.graphql.impl.GraphQLInputDeserializer.class.getName()),
+                new ReflectiveClassBuildItem(true, true, GraphQLBatch.class.getName()),
+                new ReflectiveClassBuildItem(true, true, GraphQLQuery.class.getName()));
+    }
+
+    @BuildStep
+    List<NativeImageResourceBuildItem> registerNativeImageResources() throws IOException {
+        ResolvedArtifact artifact = getVertxGraphQlArtifact();
+
+        List<NativeImageResourceBuildItem> nativeImageResourceBuildItems = new ArrayList<>();
+
+        File artifactFile = artifact.getArtifactPath().toFile();
+        try (JarFile jarFile = new JarFile(artifactFile)) {
+            Enumeration<JarEntry> entries = jarFile.entries();
+            while (entries.hasMoreElements()) {
```

Comment:
What's the impact on performance and native image size ?

Asking because all class entries end up in `nativeImageResourceBuildItems.add(new NativeImageResourceBuildItem(entry.getName()));`

Reply:
> What's the impact on performance and native image size ?
> 
> Asking because all class entries end up in `nativeImageResourceBuildItems.add(new NativeImageResourceBuildItem(entry.getName()));`

class entries are filtered out, the idea is to keep only  `.html`, `.css`, `.js`. Does it make sense?

Reply:
Ah, I missed `!` in `!entry` :)